app/api/chat/openrouter/route.ts

```diff
@@ -0,0 +1,47 @@
+import { CHAT_SETTING_LIMITS } from "@/lib/chat-setting-limits"
+import { checkApiKey, getServerProfile } from "@/lib/server/server-chat-helpers"
+import { ChatSettings } from "@/types"
+import { OpenAIStream, StreamingTextResponse } from "ai"
+import OpenAI from "openai"
+import { ChatCompletionCreateParamsBase } from "openai/resources/chat/completions.mjs"
+
+export const runtime = "edge"
+
+export async function POST(request: Request) {
+  const json = await request.json()
+  const { chatSettings, messages } = json as {
+    chatSettings: ChatSettings
+    messages: any[]
+  }
+
+  try {
+    const profile = await getServerProfile()
+
+    checkApiKey(profile.openrouter_api_key, "OpenAI")
+
+    const openai = new OpenAI({
+      apiKey: profile.openrouter_api_key || "",
+      baseURL: "https://openrouter.ai/api/v1",
+      defaultHeaders: {}
+    })
+
+    const response = await openai.chat.completions.create({
+      model: chatSettings.model as ChatCompletionCreateParamsBase["model"],
+      messages: messages as ChatCompletionCreateParamsBase["messages"],
+      temperature: chatSettings.temperature,
+      max_tokens: undefined,
+      stream: true
+    })
+
+    const stream = OpenAIStream(response)
+
+    return new StreamingTextResponse(stream)
+  } catch (error: any) {
+    const errorMessage = error.error?.message || "An unexpected error occurred"
```

Comment:
Instead of hardcoding HTTP status codes like 500, introduce a Status constant object for better code readability and maintainability. 
e.g
const Status = {
  SERVER_ERROR: 500,
};

const errorCode = error.status || Status. SERVER_ERROR

Reply:
This is actually copied from the openai endpoint I didn't even look at it.  That being said it might be worth refactoring both with your suggestions and abstracting it out into an openAIapiHandler function thats shared to keep things DRY.  I don't mind doing that but should it be in this PR or another one?

Reply:
Added an ErrorResponseSchema to parse the error and ensure proper typing